xilem/src/window_view.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright 2025 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use masonry_winit::{app::RenderRoot, widgets::RootWidget};
+use winit::window::{Window, WindowAttributes};
+use xilem_core::{AnyViewState, View, ViewElement, ViewMarker};
+
+use crate::{AnyWidgetView, ViewCtx, WindowOptions};
+
+pub(crate) struct WindowView<State> {
+    options: WindowOptions<State>,
+    root_widget_view: Box<AnyWidgetView<State, ()>>,
+}
```

Comment:
So I'm noticing that `WindowView` is more or less an analogue to `WidgetView`, but whereas `WidgetView` is a trait, `WindowView` is a type.

I think a lot of this code could get cleaner if you went the full way and created a parallel trait hierarchy:

- `trait WindowView { ... }`
- `trait WindowViewSequence { ... }`
- `struct Window { ... }`

Reply:
I don't quite see how introducing more traits would clear a lot up (see also discussion below regarding `keep_running`), could you elaborate?. Regarding ViewSequence I think that's quite unnecessary if an iterator suffices.